src/index.ts

```diff
@@ -589,15 +589,15 @@ export class AzureOpenAI extends OpenAI {
       if (!Core.isObj(options.body)) {
         throw new Error('Expected request body to be an object');
       }
-      const model = this._deployment || options.body['model'];
+      const model = this.deploymentName || options.body['model'];
       if (model !== undefined && !this.baseURL.includes('/deployments')) {
         options.path = `/deployments/${model}${options.path}`;
       }
     }
     return super.buildRequest(options, props);
   }
 
-  private async _getAzureADToken(): Promise<string | undefined> {
+  async getAzureADToken(): Promise<string | undefined> {
```

Comment:
question: do you think this is helpful to expose to users and something we should make part of the public API?

If not and this is only public for usage in the realtime client implementation then we should at least keep the `_` prefix or keep it `private` and just `ts-ignore` at the call-site